webapp/tests/Unit/Controller/API/ConfigControllerTest.php

```diff
@@ -0,0 +1,164 @@
+<?php declare(strict_types=1);
+
+namespace App\Tests\Unit\Controller\API;
+
+class ConfigControllerTest extends BaseTest
+{
+    private $endpoint = '/config';
+
+    /**
+     * Test that same public config variables are returned for all role types
+     */
+    public function testConfigReturnsPublicVariables()
+    {
+        foreach([null, 'demo', 'admin'] as $user) {
```

Comment:
I personally prefer the @dataProvider here so its easy to see for which role the test fails,

Reply:
Agreed (or add a message with the role to the asserts)